packages/nimble-components/src/wafer-map/index.ts

```diff
@@ -225,40 +246,33 @@ export class WaferMap<
      * The updates snowball one after the other, this function only choses the 'altitude'.
      * The hover does not require an event update, but it's also the last update in the sequence.
      */
-    public update(): void {
+    public async update(): Promise<void> {
+        this.zoomHandler.connect();
         this.validate();
         if (this.validity.invalidDiesTableSchema) {
             return;
         }
         this.renderer = this.isExperimentalRenderer()
             ? this.workerRenderer
             : this.mainRenderer;
-        if (this.waferMapUpdateTracker.requiresEventsUpdate) {
-            // zoom translateExtent needs to be recalculated when canvas size changes
-            this.zoomHandler.disconnect();
-            this.dataManager = this.isExperimentalRenderer()
-                ? this.experimentalDataManager
-                : this.stableDataManager;
-            if (this.waferMapUpdateTracker.requiresContainerDimensionsUpdate) {
-                this.dataManager.updateContainerDimensions();
-                this.renderer.updateSortedDiesAndDrawWafer();
-            } else if (this.waferMapUpdateTracker.requiresScalesUpdate) {
-                this.dataManager.updateScales();
-                this.renderer.updateSortedDiesAndDrawWafer();
-            } else if (
-                this.waferMapUpdateTracker.requiresLabelsFontSizeUpdate
-            ) {
-                this.dataManager.updateLabelsFontSize();
-                this.renderer.updateSortedDiesAndDrawWafer();
-            } else if (
-                this.waferMapUpdateTracker.requiresDiesRenderInfoUpdate
-            ) {
-                this.dataManager.updateDiesRenderInfo();
-                this.renderer.updateSortedDiesAndDrawWafer();
-            } else if (this.waferMapUpdateTracker.requiresDrawnWaferUpdate) {
-                this.renderer.drawWafer();
-            }
-            this.zoomHandler.connect();
+        // zoom translateExtent needs to be recalculated when canvas size changes
+        this.dataManager = this.isExperimentalRenderer()
+            ? this.experimentalDataManager
+            : this.stableDataManager;
+        if (this.waferMapUpdateTracker.requiresContainerDimensionsUpdate) {
+            this.dataManager.updateContainerDimensions();
+            await this.renderer.updateSortedDiesAndDrawWafer();
+        } else if (this.waferMapUpdateTracker.requiresScalesUpdate) {
```

Comment:
Is it appropriate for this to be all `else if` blocks? Is it possible for there to be more than one aspect to be tracked? Generally this is expected to be the case, which is why we needed an update tracker, because more than one thing can change state that should both be processed at the same time.

Reply:
Discussed in private